lib/robot.js

```diff
@@ -56,60 +73,6 @@ class Robot {
     }
   }
 
-  /**
-   * Listen for [GitHub webhooks](https://developer.github.com/webhooks/),
-   * which are fired for almost every significant action that users take on
-   * GitHub.
-   *
-   * @param {string} event - the name of the [GitHub webhook
-   * event](https://developer.github.com/webhooks/#events). Most events also
-   * include an "action". For example, the * [`issues`](
-   * https://developer.github.com/v3/activity/events/types/#issuesevent)
-   * event has actions of `assigned`, `unassigned`, `labeled`, `unlabeled`,
-   * `opened`, `edited`, `milestoned`, `demilestoned`, `closed`, and `reopened`.
-   * Often, your bot will only care about one type of action, so you can append
-   * it to the event name with a `.`, like `issues.closed`.
-   *
-   * @param {Robot~webhookCallback} callback - a function to call when the
-   * webhook is received.
-   *
-   * @example
-   *
-   * robot.on('push', context => {
-   *   // Code was just pushed.
-   * });
-   *
-   * robot.on('issues.opened', context => {
-   *   // An issue was just opened.
-   * });
-   */
-  on (event, callback) {
-    if (event.constructor === Array) {
-      event.forEach(e => this.on(e, callback))
-      return
-    }
-
-    const [name, action] = event.split('.')
-
-    return this.events.on(name, async event => {
-      if (!action || action === event.payload.action) {
-        const log = this.log.child({id: event.id})
-
-        try {
-          const github = await this.auth(event.payload.installation.id, log)
-          const context = new Context(event, github, log)
-
-          await callback(context)
-        } catch (err) {
-          log.error({err, event})
-          if (!this.catchErrors) {
-            throw err
```

Comment:
In test environments, it's really nice if these errors could bubble up to cause the tests to actually fail.